[#13964 #13962] Display contact requests

ilmotta opened the pull request:
fixes #13962
fixes #13964

### Summary

This PR adds support for Contact Request notifications in the new Activity Center (read-only at the moment, i.e. non-interactive).

<table>
  <thead>
    <tr>
      <th>
        <img alt="Screenshot" src="https://user-images.githubusercontent.com/46027/190483706-ec2f9ccf-c9e3-497e-9f8a-8a2ad66f2eee.png" height="600"/>
        <p><strong>Contact requests in various states</strong></p>
      </th>
    </tr>
  </thead>
</table>

### Review notes

- Upgrades clj-kondo to the latest version. The original version was pretty old, and it was the cause of incorrect static analysis that generated errors in the CI. For an example, see these [logs in Jenkins]( https://ci.status.im/blue/rest/organizations/jenkins/pipelines/status-mobile/pipelines/prs/pipelines/tests/branches/PR-14014/runs/4/nodes/35/log/?start=0).
- The user can't yet interact with notifications. See some of the remaining issues in https://github.com/status-im/status-mobile/labels/Activity%20center.
- The new `status-tag` component was improved to accept custom labels, otherwise it was only capable of displaying fixed text, like "positive", "negative", etc.
- Figma flows https://www.figma.com/file/eDfxTa9IoaCMUy5cLTp0ys/Shell-for-Mobile?node-id=639%3A21926
- The `utils.datetime` namespace was reformatted to use the widely accepted convention in Lisps (and Clojure) to separate top-level sections with 3 or 4 backticks. See (https://github.com/bbatsov/clojure-style-guide#heading-comments). On a side note, if you use Emacs, you can enable the `outline-minor-mode` to automatically get collapsible sections (as in Org mode).

<table>
  <thead>
    <tr>
      <th>
        <img alt="Screenshot" src="https://user-images.githubusercontent.com/46027/190483131-c3f45ba5-e4f4-41bf-a842-6c6674a136f7.jpg" width="500"/>
        <p><strong>Lisp heading comments can be collapsed in Emacs</strong></p>
      </th>
    </tr>
  </thead>
</table>

### Testing notes

Just like the merged PR https://github.com/status-im/status-mobile/pull/13987, this one doesn't yet replace the old Notifications Center with the new Activity Center. This will only be done once the Status Go [draft branch](https://github.com/status-im/status-go/tree/feature/activity-center-read-unread) is merged, but there are still some outstanding tests to be done.

Meanwhile, you need to do the following:

- Checkout status-go branch `feature/activity-center-read-unread`.
- Always run `export STATUS_GO_SRC_OVERRIDE=$HOME/<path-to-your-status-go-repo>` before running targets such as `make run-android`.
- Apply the following patch to status-mobile:

```diff
modified   src/status_im/ui/screens/home/views.cljs
@@ -270,7 +270,7 @@
                           :accessibility-label :notifications-button
                           :on-press #(do
                                        (re-frame/dispatch [:mark-all-activity-center-notifications-as-read])
-                                       (re-frame/dispatch [:navigate-to :notifications-center]))}
+                                       (re-frame/dispatch [:navigate-to :activity-center]))}
       [icons/icon :main-icons/notification2 {:color (quo2.colors/theme-colors quo2.colors/black quo2.colors/white)}]]
      (when (pos? notif-count)
        [react/view {:style (merge (styles/counter-public-container) {:top 5 :right 5})
```

### Steps to test

You can follow the use cases below if you'd like to do some manual testing. Please, don't consider this an exaustive list of use cases. In all cases, Bob has enabled _accept chat from anyone_ in his app settings.

>Given that Alice has not added Bob as a contact,
>When Alice opens a chat with Bob by using his chat key,
>And Alice sends her first message to him,
>Then Bob sees the last message from Alice in his Activity Center (aka 1:1 notification).

>Given that Alice already sent a message to Bob,
>When Alice opens the chat with Bob,
>And when Alice clicks on Add Contact,
>Then Bob receives a notification in the Activity Center,
>And Bob can either accept or decline the contact request.

>Given that Bob accepted a contact request from Alice,
>When Bob removes Alice from his contact list,
>Then Bob doesn't see notifications related to Alice in the Activity Center.

>Given that Bob accepted a contact request from Alice,
>When Bob opens the Activity Center,
>Then he sees a notification saying he accepted the contact request from Alice.

>Given that Bob declined a contact request from Alice,
>When Bob opens the Activity Center,
>Then he sees a notification saying he declined the contact request from Alice.

>Given that Bob gave a nickname to Alice (only he sees),
>When Bob opens the Activity Center,
>Then he sees all references to Alice replaced with her nickname.

status: ready

ilmotta committed:
Display contact requests

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 79baddf1 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14014/1/) | 2022-09-15 18:55:57 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14014/1/consoleText) |

ilmotta committed:
Dummy commit to see if build cache is affected

ilmotta committed:
Upgrade clj-kondo to latest version

This was due to a weird bug, where clj-kondo would report a namespace wasn't
used, even though it clearly was.

ilmotta commented on src/status_im/ui/screens/activity_center/views.cljs:
```diff
@@ -5,26 +5,88 @@
+    constants/activity-center-notification-type-one-to-one-chat
+    "Dummy 1:1 chat title"
+
+    "Dummy default title"))
```
FYI: More cases for different notification types will be covered as the new Activity Center issues are played. This is true for this function and for others in this namespace.

flexsurfer commented on src/quo2/screens/tags/status_tags.cljs:
```diff
@@ -27,16 +27,20 @@
-                 :justify-content  :center}
-        [quo2/status-tag @state]]])))
+      (let [props (cond-> @state
+                    (= :positive (:status @state)) (assoc :status {:label "Positive" :type :positive})
```
why its a duplicate value from tags-options ? also if we show this to users it must be i18n

J-Son89 replied:
I think this is okay. The `Status Tags` text was hardcoded before but needs to be dynamic. I noticed similar in some work I was doing.

And the text here is only for quo2 component library so it's probably okay to not have it i18n, right?

flexsurfer replied:
oh right , its a quo screen, thanks

J-Son89 commented on src/status_im/utils/datetime.cljs:
```diff
@@ -10,6 +10,8 @@
             [clojure.string :as s]
             [status-im.goog.i18n :as goog.18n]))
 
+;;;; Datetime constants
```
nit: are these really constants? 🤔

ilmotta replied:
You're right, most of them are constants (time units), but others are not. I would gladly take a suggestion here.

J-Son89 commented on src/status_im/utils/datetime.cljs:
```diff
@@ -29,28 +31,54 @@
+    "E h:mm a"))
+
+(defn- medium-date-format
+  "Get medium format from current locale symbols."
```
what is medium format? 🤨

ilmotta replied:
@J-Son89, if you look at the diff, you'll see that I tried only to reformat and move the original comment to the docstring. But ideally we could do a better job at describing some of these things.

J-Son89 commented on src/status_im/utils/datetime.cljs:
```diff
@@ -97,6 +119,35 @@
+        six-days-ago   (minus today (days 6))]
+    (cond
+      ;; Previous years.
+      (< (t/year datetime) (t/year today))
```
consider putting these into helper functions and remove comments.
```
(defn previous-years? [datetime today datetimelocal]
(< (t/year datetime) (t/year today))
      (.format ^js (date-fmt) datetime-local))
```
same for below.

ilmotta replied:
Good suggestion, I'll do my best to see if this can be refactored. Just so you understand, the reason why I chose to keep things like this is because every subsequent check in the `cond` macro only works because of the previous checks.

For example, the correct way to check if a datetime is yesterday is not just `(before? datetime today)`. I hope you get it. I'll see what I can come up with, thanks.

J-Son89 replied:
yeah I get that, I mean just use named functions for the checks rather than comments. You can still keep the function `defn timestamp->relative [ms]` as is

ilmotta replied:
It was quite more involved than a simple extraction, especially because I had to make sure the new implementation is correct. The new extracted functions are now useful on their own, and are correct, independent of the context they're being used, which was not the case of the original code. I also removed other superfluous timezone-related code that wasn't really necessary. I'm marking this thread as resolved :)

ilmotta committed:
Refactor timestamp->relative and add more tests

Parveshdhull commented:
> Upgrades clj-kondo to the latest version.

hi @ilmotta, Thank you for upgrading `clj-kondo` this also fixes https://github.com/status-im/status-mobile/issues/13991

Parveshdhull commented on src/quo2/components/tags/status_tags.cljs:
```diff
@@ -50,42 +50,48 @@
+  [base-tag {:size             size
              :background-color colors/success-50-opa-10
-             :icon :verified
-             :border-color colors/success-50-opa-20
-             :text-color (if (= theme :light) colors/success-50
-                             colors/success-60)
-             :label (or label (i18n/label :positive))}])
+             :icon             :verified
+             :border-color     colors/success-50-opa-20
+             :label            (or label (i18n/label :positive))
+             :text-color       (if (= theme :light) colors/success-50
+                                   colors/success-60)}])
 
```
hi @ilmotta, why these changes needed?

ilmotta replied:
@Parveshdhull, what happened here is that I had improved the status tag component to accept a label, but someone also fixed it and merged before I did. So this morning I fixed the conflicts, but these changes remained after my rebase (map keys were sorted). They're harmless though

status-im-auto commented:
## 97% of end-end tests have passed
```
Total executed tests: 87
Failed tests: 3
Passed tests: 84
```
```
IDs of failed tests: 5756,700732,3997 
```
<h3>Failed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14014.apk&PR_ID=14014&TR_CASE_IDS=5756,700732,3997">Rerun failed tests</a></li><br/><h4>Class TestGroupChatMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994916">test_group_chat_highligted</a>, id: 5756 </b></td></tr><tr><td><p><blockquote><div>Device 1: Double tap on: HomeButton </div><div>Device 1: Find `HomeButton` by `xpath`: `//*[contains(@content-desc,'tab, 1 out of 5')]` </div></blockquote></p>

```
critical/chats/test_group_chat.py:162: in test_group_chat_highligted
    [self.homes[i].home_button.double_click() for i in range(3)]
critical/chats/test_group_chat.py:162: in <listcomp>
    [self.homes[i].home_button.double_click() for i in range(3)]
../views/base_element.py:110: in double_click
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:110: in <listcomp>
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: HomeButton by xpath: `//*[contains(@content-desc,'tab, 1 out of 5')]` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/35559cb152784887a598e8ba819025c2?auth=d0e47ba7d8e841159821e358ece29d90#180">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/35559cb152784887a598e8ba819025c2/0050screenshot.png?auth=d0e47ba7d8e841159821e358ece29d90">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b44009d72ba8411c879b1860484df549?auth=6c1d32814259470c43ccb50f242ce9e2#132">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/b44009d72ba8411c879b1860484df549/0029screenshot.png?auth=6c1d32814259470c43ccb50f242ce9e2">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28?auth=889232c6c664e486f7d753a37481a58f#103">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28/0023screenshot.png?auth=889232c6c664e486f7d753a37481a58f">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994929">test_group_chat_add_new_member_activity_centre</a>, id: 700732 </b></td></tr><tr><td><p><blockquote><div>Device 3: Find `Button` by `accessibility id`: `notifications-unread-badge` </div><div>Device 3: Tap on found: Button </div></blockquote></p>

```
critical/chats/test_group_chat.py:118: in test_group_chat_add_new_member_activity_centre
    self.homes[2].get_chat(self.chat_name).click()
../views/home_view.py:257: in get_chat
    chat_in_ac.wait_for_element(20)
../views/base_element.py:117: in wait_for_element
    raise TimeoutException(
 Device `3`: `ActivityCenterChatElement` by` xpath`: `//*[@content-desc='chat-name-or-sender-text'][starts-with(@text,'ucsflka')]/../..` is not found on the screen after wait_for_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/35559cb152784887a598e8ba819025c2?auth=d0e47ba7d8e841159821e358ece29d90#150">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/35559cb152784887a598e8ba819025c2/0050screenshot.png?auth=d0e47ba7d8e841159821e358ece29d90">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b44009d72ba8411c879b1860484df549?auth=6c1d32814259470c43ccb50f242ce9e2#122">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/b44009d72ba8411c879b1860484df549/0029screenshot.png?auth=6c1d32814259470c43ccb50f242ce9e2">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28?auth=889232c6c664e486f7d753a37481a58f#77">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28/0023screenshot.png?auth=889232c6c664e486f7d753a37481a58f">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994902">test_group_chat_leave_relogin</a>, id: 3997 </b></td></tr><tr><td><p><blockquote><div>Device 1: Double tap on: HomeButton </div><div>Device 1: Find `HomeButton` by `xpath`: `//*[contains(@content-desc,'tab, 1 out of 5')]` </div></blockquote></p>

```
critical/chats/test_group_chat.py:178: in test_group_chat_leave_relogin
    [self.homes[i].home_button.double_click() for i in range(2)]
critical/chats/test_group_chat.py:178: in <listcomp>
    [self.homes[i].home_button.double_click() for i in range(2)]
../views/base_element.py:110: in double_click
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:110: in <listcomp>
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: HomeButton by xpath: `//*[contains(@content-desc,'tab, 1 out of 5')]` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/35559cb152784887a598e8ba819025c2?auth=d0e47ba7d8e841159821e358ece29d90#184">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/35559cb152784887a598e8ba819025c2/0050screenshot.png?auth=d0e47ba7d8e841159821e358ece29d90">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b44009d72ba8411c879b1860484df549?auth=6c1d32814259470c43ccb50f242ce9e2#135">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/b44009d72ba8411c879b1860484df549/0029screenshot.png?auth=6c1d32814259470c43ccb50f242ce9e2">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28?auth=889232c6c664e486f7d753a37481a58f#106">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28/0023screenshot.png?auth=889232c6c664e486f7d753a37481a58f">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (84)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestPublicChatMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994908">test_public_chat_delete_chat_long_press</a>, id: 5319 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#231">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#157">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994920">test_public_chat_mark_all_messages_as_read</a>, id: 6270 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#320">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#284">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994930">test_public_chat_message_edit</a>, id: 700734 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#119">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#76">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994931">test_public_chat_message_delete</a>, id: 700735 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#134">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#85">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994927">test_public_chat_emoji_send_copy_paste_reply</a>, id: 700719 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#147">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#89">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994910">test_public_chat_unread_messages_counter</a>, id: 5360 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#174">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#102">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994932">test_public_chat_link_send_open</a>, id: 700736 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#266">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#193">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994926">test_public_chat_unread_messages_counter_for_mention_relogin</a>, id: 700718 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#200">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#110">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994905">test_public_chat_message_send_check_timestamps_while_on_different_tab</a>, id: 5313 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#101">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#58">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994933">test_public_chat_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 700737 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a0ad64b2ebf409a9306bce1205cb618?auth=67d4a21ef9e97f8e73dbeb7186a66ddc#283">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8e2bd55372684ea1b09ac607a003c7fc?auth=db3d2b56fea7231fb51af43e4a7fadbf#217">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestRestoreOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994978">test_restore_account_migrate_multiaccount_to_keycard_no_db_saved_add_wallet_send_tx</a>, id: 702189 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3d986901bd06409784b0189eb14ccb76?auth=22bf09b5a05dd81b336d615084037a8d#242">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994944">test_restore_seed_phrase_field_validation</a>, id: 700750 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3d986901bd06409784b0189eb14ccb76?auth=22bf09b5a05dd81b336d615084037a8d#172">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994943">test_restore_set_up_wallet_sign_phrase</a>, id: 700749 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3d986901bd06409784b0189eb14ccb76?auth=22bf09b5a05dd81b336d615084037a8d#67">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994942">test_restore_uppercase_whitespaces_seed_phrase_special_char_passw_logcat</a>, id: 700748 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3d986901bd06409784b0189eb14ccb76?auth=22bf09b5a05dd81b336d615084037a8d#42">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestEnsStickersMultipleDevicesMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994967">test_ens_purchased_in_profile</a>, id: 702152 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/be159e13e4e24c1396b450c11f54a146?auth=3f17b12eb0d4e90a439680320a0bd498#92">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8ad51fda23a643adba657373d6cb8627?auth=9e60a16455d29aa706f4253d15c506f1#123">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994972">test_start_new_chat_public_key_validation</a>, id: 702158 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/be159e13e4e24c1396b450c11f54a146?auth=3f17b12eb0d4e90a439680320a0bd498#359">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8ad51fda23a643adba657373d6cb8627?auth=9e60a16455d29aa706f4253d15c506f1#395">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994970">test_ens_mention_push_highlighted_public_chat</a>, id: 702156 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/be159e13e4e24c1396b450c11f54a146?auth=3f17b12eb0d4e90a439680320a0bd498#254">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8ad51fda23a643adba657373d6cb8627?auth=9e60a16455d29aa706f4253d15c506f1#186">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994971">test_sticker_1_1_public_chat_mainnet</a>, id: 702157 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/be159e13e4e24c1396b450c11f54a146?auth=3f17b12eb0d4e90a439680320a0bd498#281">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8ad51fda23a643adba657373d6cb8627?auth=9e60a16455d29aa706f4253d15c506f1#225">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994969">test_ens_mention_nickname_1_1_chat</a>, id: 702155 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/be159e13e4e24c1396b450c11f54a146?auth=3f17b12eb0d4e90a439680320a0bd498#176">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8ad51fda23a643adba657373d6cb8627?auth=9e60a16455d29aa706f4253d15c506f1#170">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994968">test_ens_command_send_tx_eth_1_1_chat</a>, id: 702153 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/be159e13e4e24c1396b450c11f54a146?auth=3f17b12eb0d4e90a439680320a0bd498#95">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8ad51fda23a643adba657373d6cb8627?auth=9e60a16455d29aa706f4253d15c506f1#151">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994901">test_group_chat_push_system_messages_when_invited</a>, id: 3994 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/35559cb152784887a598e8ba819025c2?auth=d0e47ba7d8e841159821e358ece29d90#126">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b44009d72ba8411c879b1860484df549?auth=6c1d32814259470c43ccb50f242ce9e2#76">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28?auth=889232c6c664e486f7d753a37481a58f#64">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994903">test_group_chat_offline_pn</a>, id: 3998 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/35559cb152784887a598e8ba819025c2?auth=d0e47ba7d8e841159821e358ece29d90#176">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b44009d72ba8411c879b1860484df549?auth=6c1d32814259470c43ccb50f242ce9e2#129">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28?auth=889232c6c664e486f7d753a37481a58f#100">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994928">test_group_chat_join_send_text_messages_push</a>, id: 700731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/35559cb152784887a598e8ba819025c2?auth=d0e47ba7d8e841159821e358ece29d90#135">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b44009d72ba8411c879b1860484df549?auth=6c1d32814259470c43ccb50f242ce9e2#99">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/f1cbb0d0e23040e2b4765ac62e5a2b28?auth=889232c6c664e486f7d753a37481a58f#70">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevices:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994924">test_1_1_chat_audio_message_with_push</a>, id: 6316 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#90">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#85">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994912">test_1_1_chat_delete_via_delete_button_relogin</a>, id: 5387 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#468">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#415">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994906">test_1_1_chat_non_latin_message_to_newly_added_contact_with_profile_picture_on_different_networks</a>, id: 5315 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#211">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#192">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994922">test_1_1_chat_image_send_save_reply</a>, id: 6305 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#332">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#279">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994921">test_1_1_chat_push_emoji</a>, id: 6283 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#303">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#237">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994904">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 5310 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#398">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#355">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994911">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 5373 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#108">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#118">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994923">test_1_1_chat_message_reaction</a>, id: 6315 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#70">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#64">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994925">test_1_1_chat_text_message_edit_delete_push_disappear</a>, id: 695843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3204b368d4de4f19b5260d1ca6ed6ca7?auth=bb9e67a4e140d7f2a4ba393e87b104f5#153">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/59ba17f8a2c64034a9e777547f085411?auth=f6afa5bf726c33761aa6e6fbae96a5f4#141">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestKeycardTxOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994958">test_keycard_send_tx_sign_message_request_stt_testdapp</a>, id: 700769 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cf9c5525f650440a96724deef67a979c?auth=ad49b458a06ace5278ca09397038fcef#296">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994956">test_keycard_send_tx_eth</a>, id: 700767 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cf9c5525f650440a96724deef67a979c?auth=ad49b458a06ace5278ca09397038fcef#146">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994915">test_keycard_create_account_unlock_same_seed</a>, id: 5689 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cf9c5525f650440a96724deef67a979c?auth=ad49b458a06ace5278ca09397038fcef#669">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994959">test_keycard_wallet_recover_pairing_check_balance_after_offline_tx_history</a>, id: 700770 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cf9c5525f650440a96724deef67a979c?auth=ad49b458a06ace5278ca09397038fcef#536">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994957">test_keycard_relogin_after_restore</a>, id: 700768 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cf9c5525f650440a96724deef67a979c?auth=ad49b458a06ace5278ca09397038fcef#244">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestPublicChatBrowserOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994966">test_browser_web3_permissions_testdapp</a>, id: 702078 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#593">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994913">test_public_chat_navigate_to_chat_when_relaunch</a>, id: 5396 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#137">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994962">test_browser_invalid_url</a>, id: 702074 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#361">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994907">test_public_chat_copy_and_paste_message_in_chat_input</a>, id: 5317 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#169">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994914">test_public_chat_fetch_more_history</a>, id: 5675 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#54">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994935">test_public_chat_open_using_deep_link</a>, id: 700739 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#239">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994961">test_browser_connection_is_secure_not_secure_warning</a>, id: 702073 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#308">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994934">test_public_chat_tag_message</a>, id: 700738 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#208">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994965">test_browser_bookmarks_create_edit_remove</a>, id: 702077 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#500">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994963">test_browser_offline</a>, id: 702075 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#375">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994960">test_browser_blocked_url</a>, id: 702072 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#257">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>12. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994964">test_browser_delete_close_tabs</a>, id: 702076 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a3238f5223d84546b60330e9ad584738?auth=31daf50a7c79943c967ad3d1d0cceea1#419">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletManagementDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994955">test_wallet_fetching_balance_after_offline_insufficient_funds_errors</a>, id: 700766 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6675a3e4776f4fa7bd42873c63373577?auth=0af9a7d0312160ca8304daf9504ac06a#768">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994945">test_wallet_tx_history_copy_tx_hash_on_cellular</a>, id: 700756 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6675a3e4776f4fa7bd42873c63373577?auth=0af9a7d0312160ca8304daf9504ac06a#133">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994950">test_wallet_add_hide_unhide_account_private_key</a>, id: 700761 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6675a3e4776f4fa7bd42873c63373577?auth=0af9a7d0312160ca8304daf9504ac06a#496">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994949">test_wallet_add_delete_watch_only_account</a>, id: 700760 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6675a3e4776f4fa7bd42873c63373577?auth=0af9a7d0312160ca8304daf9504ac06a#351">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994947">test_wallet_manage_assets</a>, id: 700758 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6675a3e4776f4fa7bd42873c63373577?auth=0af9a7d0312160ca8304daf9504ac06a#288">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994951">test_wallet_add_account_seed_phrase_validation</a>, id: 700762 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6675a3e4776f4fa7bd42873c63373577?auth=0af9a7d0312160ca8304daf9504ac06a#731">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994948">test_wallet_add_account_generate_new</a>, id: 700759 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6675a3e4776f4fa7bd42873c63373577?auth=0af9a7d0312160ca8304daf9504ac06a#234">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommandsMultipleDevicesMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994917">test_1_1_chat_command_send_tx_eth_outgoing_tx_push</a>, id: 6253 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/d6dd2590723d4c9a894dc30824abcc39?auth=f33ee7402de91f71d260816aed52868d#87">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b546c69dab674a28bf3c39f760b2e80b?auth=c065a3c43d4c6ff5df0b87182216311e#73">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994919">test_1_1_chat_command_decline_eth_push_changing_state</a>, id: 6265 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/d6dd2590723d4c9a894dc30824abcc39?auth=f33ee7402de91f71d260816aed52868d#171">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b546c69dab674a28bf3c39f760b2e80b?auth=c065a3c43d4c6ff5df0b87182216311e#158">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994918">test_1_1_chat_command_request_and_send_tx_stt_in_1_1_chat_offline</a>, id: 6263 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/d6dd2590723d4c9a894dc30824abcc39?auth=f33ee7402de91f71d260816aed52868d#221">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/b546c69dab674a28bf3c39f760b2e80b?auth=c065a3c43d4c6ff5df0b87182216311e#180">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestSendTxDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994954">test_send_tx_custom_token_18_decimals_invalid_password</a>, id: 700765 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ee2317db060440dea56343b30aa8572f?auth=315550f042ec0ec1c272c5cc22c9f2b9#485">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994946">test_send_tx_set_recipient_options</a>, id: 700757 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ee2317db060440dea56343b30aa8572f?auth=315550f042ec0ec1c272c5cc22c9f2b9#571">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994909">test_send_tx_sign_message_2tx_in_batch_tx_filters_request_stt_testdapp</a>, id: 5342 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ee2317db060440dea56343b30aa8572f?auth=315550f042ec0ec1c272c5cc22c9f2b9#302">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994953">test_send_tx_token_8_decimals</a>, id: 700764 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ee2317db060440dea56343b30aa8572f?auth=315550f042ec0ec1c272c5cc22c9f2b9#264">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994952">test_send_tx_eth_check_logcat</a>, id: 700763 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ee2317db060440dea56343b30aa8572f?auth=315550f042ec0ec1c272c5cc22c9f2b9#222">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestContactBlockMigrateKeycardMultipleSharedDevices:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994974">test_contact_block_unblock_public_chat_offline</a>, id: 702176 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/62dc4fd657564669b81dd11f5cedf1d5?auth=3136f5d2e9daca5357489690c0a01c37#367">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/90db639dfa374cc0ae97d68b5f15c8df?auth=c921b8ad12afafe4704d99bc7517721a#242">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994976">test_keycard_command_send_tx_eth_1_1_chat</a>, id: 702186 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/62dc4fd657564669b81dd11f5cedf1d5?auth=3136f5d2e9daca5357489690c0a01c37#135">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/90db639dfa374cc0ae97d68b5f15c8df?auth=c921b8ad12afafe4704d99bc7517721a#173">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994973">test_contact_add_remove_mention_default_username_nickname_public_chat</a>, id: 702175 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/62dc4fd657564669b81dd11f5cedf1d5?auth=3136f5d2e9daca5357489690c0a01c37#229">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/90db639dfa374cc0ae97d68b5f15c8df?auth=c921b8ad12afafe4704d99bc7517721a#235">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994975">test_restore_account_migrate_multiaccount_to_keycard_db_saved</a>, id: 702177 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/62dc4fd657564669b81dd11f5cedf1d5?auth=3136f5d2e9daca5357489690c0a01c37#577">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/90db639dfa374cc0ae97d68b5f15c8df?auth=c921b8ad12afafe4704d99bc7517721a#395">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994977">test_cellular_settings_on_off_public_chat_fetching_history</a>, id: 702188 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/62dc4fd657564669b81dd11f5cedf1d5?auth=3136f5d2e9daca5357489690c0a01c37#513">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/90db639dfa374cc0ae97d68b5f15c8df?auth=c921b8ad12afafe4704d99bc7517721a#291">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOnboardingOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994936">test_onboarding_home_initial_popup</a>, id: 700742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a0d4327ff74944ba896cb37e7dea8cf5?auth=f8187703a83166a756d57070e87d84db#60">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994941">test_onboarding_add_new_multiaccount_username_by_position_pass_validation</a>, id: 700747 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a0d4327ff74944ba896cb37e7dea8cf5?auth=f8187703a83166a756d57070e87d84db#386">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994940">test_onboarding_cant_sign_in_with_invalid_password_logcat</a>, id: 700746 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a0d4327ff74944ba896cb37e7dea8cf5?auth=f8187703a83166a756d57070e87d84db#349">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994938">test_onboarding_share_wallet_address</a>, id: 700744 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a0d4327ff74944ba896cb37e7dea8cf5?auth=f8187703a83166a756d57070e87d84db#171">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994939">test_onboarding_backup_seed_phrase_restore_same_login_logcat</a>, id: 700745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a0d4327ff74944ba896cb37e7dea8cf5?auth=f8187703a83166a756d57070e87d84db#246">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994937">test_onboarding_share_contact_address</a>, id: 700743 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a0d4327ff74944ba896cb37e7dea8cf5?auth=f8187703a83166a756d57070e87d84db#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestPairingSyncMultipleDevicesMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994984">test_pairing_sync_public_chat_add_remove</a>, id: 702199 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#511">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#166">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994986">test_pairing_sync_initial_bookmarks</a>, id: 702393 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#562">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#192">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994980">test_pairing_sync_initial_public_chats</a>, id: 702195 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#344">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994983">test_pairing_sync_1_1_chat_message</a>, id: 702198 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#493">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#149">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994982">test_pairing_sync_contacts_add_remove_set_nickname_ens</a>, id: 702197 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#414">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#127">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994987">test_pairing_sync_clear_history</a>, id: 702394 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#569">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#204">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994979">test_pairing_sync_initial_contacts_blocked_users</a>, id: 702194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#341">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#85">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994985">test_pairing_sync_initial_profile_picture</a>, id: 702392 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#555">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#178">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2994981">test_pairing_sync_contacts_block_unblock</a>, id: 702196 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4e2a6a65e6c94675b6518f9f335a91bf?auth=e4e22d849b53f2dd0c55613e225c2abc#351">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f2d0f819cd3f40ac9e46a1807470f264?auth=b8812e455edb8ed18282295534e58425#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>
